core/lib/zksync_core/src/native_token_fetcher/mod.rs

```diff
@@ -70,12 +72,15 @@ pub(crate) struct NativeTokenFetcher {
 
 impl NativeTokenFetcher {
     pub(crate) async fn new(config: NativeTokenFetcherConfig) -> Self {
-        let conversion_rate = reqwest::get(format!("{}/conversion_rate", config.host))
-            .await
-            .unwrap()
-            .json::<u64>()
-            .await
-            .unwrap();
+        let conversion_rate = match reqwest::get(format!("{}/conversion_rate", config.host)).await {
+            Ok(response) => response.json::<u64>().await.unwrap_or(1), // Prevent program from crashing altogether if the first request fails
+            Err(err) => {
+                tracing::error!(
+                    "Failed to fetch native token conversion rate from the server: {err}"
+                );
+                1
+            }
+        };
```

Comment:
I'm not 100% sure about that. When I wrote about [running with unsynchronized ratios](https://github.com/matter-labs/zksync-era/pull/996#discussion_r1503677218) I was thinking about using some outdated, but still real value (e.g. persisting the history of fetched ratios) -- which is why I wrote that we can tackle it separately, since it's a big topic.

What you proposed in the same discussion -- panic (or, even better, return `anyhow::Result<Self>`) if the reliable rate cannot be fetcher for the first time -- seems to be a better short-term approach.

Reply:
Maybe we can just `.process()` this possible initial with the new `ErrorReporter` structure discussed above.